packages/embark/src/lib/modules/coverage/contractEnhanced.ts

```diff
@@ -32,13 +32,14 @@ export class ContractEnhanced {
 
   constructor(public filepath: string, public solcVersion: string) {
     this.id = nextId();
+    this.source = fs.readFileSync(filepath, "utf-8");
+    this.originalSource = this.source;
+
     try {
-      this.source = fs.readFileSync(filepath, "utf-8");
-      this.originalSource = this.source;
       this.ast = parser.parse(this.source, {loc: true, range: true});
     } catch (error) {
-      this.source = "";
-      this.originalSource = "";
+      const {line, column, message} = error.errors[0];
+      console.warn(`Error on ${this.filepath}:${line}:${column}: "${message}". Could not setup for coverage.`);
```

Comment:
I am not a super big fan of doing this here, but alas, the constructor is already doing work, which it shouldn't, so we'll leave it the way it is for now.